osu.Game.Tests/Skins/TestSceneSkinDeserialisation.cs

```diff
@@ -0,0 +1,194 @@
+            // Covers "Argon" health display
+            "Archives/modified-argon-pro-20231001.osk",
+            // Covers player name text component.
+            "Archives/modified-argon-20231106.osk",
+            // Covers "Argon" accuracy/score/combo counters, and wedges
+            "Archives/modified-argon-20231108.osk",
+            // Covers "Argon" performance points counter
+            "Archives/modified-argon-20240305.osk",
+            // Covers default rank display
+            "Archives/modified-default-20230809.osk",
+            // Covers legacy rank display
+            "Archives/modified-classic-20230809.osk",
+            // Covers legacy key counter
+            "Archives/modified-classic-20240724.osk"
+        };
+
+        [Resolved]
+        private SkinManager skins { get; set; } = null!;
+
+        /// <summary>
+        /// If this test fails, new test resources should be added to include new components.
+        /// </summary>
+        [Test]
+        public void TestSkinnableComponentsCoveredByDeserialisationTests()
+        {
+            HashSet<Type> instantiatedTypes = new HashSet<Type>();
+
+            AddStep("load skin", () =>
+            {
+                foreach (string oskFile in available_skins)
+                {
+                    var skin = importSkinFromArchives(oskFile);
+
+                    foreach (var target in skin.LayoutInfos)
+                    {
+                        foreach (var info in target.Value.AllDrawables)
+                            instantiatedTypes.Add(info.Type);
+                    }
+                }
+            });
+
+            var existingTypes = SerialisedDrawableInfo.GetAllAvailableDrawables().Where(t => (Activator.CreateInstance(t) as ISerialisableDrawable)?.IsEditable == true);
+
+            AddAssert("all types available", () => instantiatedTypes, () => Is.EquivalentTo(existingTypes));
+        }
+
+        [Test]
+        public void TestDeserialiseModifiedDefault()
+        {
+            Skin skin = null!;
+
+            AddStep("load skin", () => skin = importSkinFromArchives("Archives/modified-default-20220723.osk"));
+
+            AddAssert("layouts count = 2", () => skin.LayoutInfos, () => Has.Count.EqualTo(2));
+            AddAssert("hud count = 12",
+                () => skin.LayoutInfos[GlobalSkinnableContainers.MainHUDComponents].AllDrawables.ToArray(),
+                () => Has.Length.EqualTo(12));
+        }
+
+        [Test]
+        public void TestDeserialiseModifiedArgon()
+        {
+            Skin skin = null!;
+
+            AddStep("load skin", () => skin = importSkinFromArchives("Archives/modified-argon-20231106.osk"));
+
+            AddAssert("layouts count = 2", () => skin.LayoutInfos, () => Has.Count.EqualTo(2));
+            AddAssert("hud count = 13",
+                () => skin.LayoutInfos[GlobalSkinnableContainers.MainHUDComponents].AllDrawables.ToArray(),
+                () => Has.Length.EqualTo(13));
```

Comment:
Some may notice the count assertions on some of these tests have changed. This is because previously the test did not count the combo counter migration, which moves the combo counter from the global layout to each ruleset's layout, therefore increasing the `AllDrawables` count by 3.